src/Lua.cpp

```diff
@@ -97,7 +97,7 @@ static void get_host_vlan_info(char* lua_ip, char** host_ip,
   if(((*host_ip) = strtok_r(buf, "@", &where)) != NULL)
     vlan = strtok_r(NULL, "@", &where);
 
-  if(host_ip == NULL)
+  if(*host_ip == NULL)
```

Comment:
this is wrong, host_ip is a pointer to a pointer, and you want to check if it points to a NULL address before assigning to that pointed address.